drgn_tools/multipath.py

```diff
@@ -0,0 +1,330 @@
+    """
+    header_mapping = {
+        "service-time": "   InFlightIO_Size   RThroughput  RCount  Load",
+        "round-robin": "   Repeat_Count",
+        "queue-length": "   Repeat_Count  Queue_Length",
+        "io-affinity": "   Refcount   failed",
+        "historical-service-time": "Feature not supported for "
+        "historical-service-time.",
+    }
+    header = header_mapping.get(ps_type)
+    if header is not None:
+        print(header)
+    else:
+        """
+        We shouldn't reach here.
+        """
+        print("\n\nPath selection policy %s not supported.\n\n" % ps_type)
+
+
+def switch_output(ps_type: str, pi: Object) -> None:
+    """
+    Format the path info output as per ps policy.
+
+    :param  ps_type: Path selection policy name.
+    :param pi: ``void *`` of path_info (must be decoded by the policy)
+
+    :return None.
+    """
+    output_mapping = {
+        "service-time": "{:<6}{:<12d}{:<5}{:<7d}{:<3}{:<6d}{:<1}{:d}",
+        "round-robin": "{:<5}{:d}",
+        "queue-length": "{:<5}{:<9d}{:<5}{:d}",
+        "io-affinity": "{:<4}{:<7d}{:<3}{:d}",
+        "historical-service-time": "",
+    }
+    output_format = output_mapping.get(ps_type)
+    if not output_format:
+        print("\n\nPath selection policy %s not supported.\n\n" % ps_type)
+        return
+    if ps_type == "service-time":
+        load = pi.in_flight_size.counter / pi.relative_throughput
+        print(
+            output_format.format(
+                " ",
+                pi.in_flight_size.counter.value_(),
+                " ",
+                pi.relative_throughput.value_(),
+                " ",
+                pi.repeat_count.value_(),
+                " ",
+                load.value_(),
+            )
+        )
+    elif ps_type == "round-robin":
+        print(output_format.format(" ", pi.repeat_count.value_()))
+    elif ps_type == "queue-length":
+        print(
+            output_format.format(
+                " ",
+                pi.repeat_count.value_(),
+                " ",
+                pi.qlen.counter.value_(),
+            )
+        )
+    elif ps_type == "io-affinity":
+        print(
+            output_format.format(
+                " ",
+                pi.refcount.refs.counter.value_(),
+                " ",
+                pi.failed.value_(),
+            )
+        )
+    else:
+        print(output_format)
+
+
+def get_path_info(prog: Program, ps_type: str) -> Type:
+    ps_type_us = ps_type.replace("-", "_")
+    return type_lookup_conflict(
+        prog,
+        "struct path_info *",
+        f"dm_{ps_type_us}",  # module: e.g. dm_service_time
+        [f"dm-{ps_type}.c", f"dm-ps-{ps_type}.c"],  # e.g. dm-service-time.c
+    )
+
+
+def show_pgpaths(priority_group: Object) -> None:
+    """
+    Show pgpath data.
+
+    :param priority_group: `struct priority_group *`
+
+    :returns: None
+    """
+    path_num = 1
+    ps_type = priority_group.ps.type.name.string_().decode()
```

Comment:
I think this function should use ensure_debuginfo() to make sure the required mode for the path selector is loaded. Corelens will make sure the modules list in Multipath.debuginfo_kmods be loaded, but if users invokes the helpes in interactive mode, this function will fail due to modules not loaded.